flake8_typing_imports.py

```diff
@@ -447,18 +449,52 @@ def __init__(self) -> None:
         self.imports: Dict[str, List[Tuple[int, int]]]
         self.imports = collections.defaultdict(list)
         self.defined_overload = False
+        self.unions_pattern_or_match: List[Tuple[int, int]] = []
+        self.from_imported_names: Set[str] = set()
+
+    def _is_typing(self, node: ast.AST, names: Container[str]) -> bool:
+        return (
+            isinstance(node, ast.Name) and
+            node.id in names and
+            node.id in self.from_imported_names
+        ) or (
+            isinstance(node, ast.Attribute) and
+            isinstance(node.value, ast.Name) and
+            node.value.id == 'typing' and
+            node.attr in names
+        )
 
     def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
-        if node.module == 'typing' and self._level == 0:
```

Comment:
I think you can keep this logic

Reply:
Do you mean it should be changed to this?
```python
if node.module == 'typing' and self._level == 0:
    for name in node.names:
        self.imports[name.name].append((node.lineno, node.col_offset))
        if not name.asname:
            self.from_imported_names.add(name.name)

```

Reply:
yeah